awscfncli2/cli/utils/deco.py

```diff
@@ -29,3 +30,35 @@ def wrapper(ctx, *args, **kwargs):
             raise click.Abort
 
     return wrapper
+
+
+def retry_on_throttling(tries=4, delay=2, backoff=2):
+    """
+    Retries function on Throttling errors raised by WaiterError or ClientError
+
+    Args:
+        tries: number of attempts before raising any exceptions
+        delay: initial delay between retries in seconds
+        backoff: backoff multiplier e.g. value of 2 will double the delay each retry
+    """
+
+    def decorator_retry(f):
+        @wraps(f)
+        def f_retry(*args, **kwargs):
+            mtries, mdelay = tries, delay
+            while mtries > 1:
+                try:
+                    return f(*args, **kwargs)
+                except (botocore.exceptions.WaiterError, botocore.exceptions.ClientError) as e:
+                    if 'Rate exceeded' in str(e):
```

Comment:
I'm wondering is there a specific boto3 exception or the rate limit can only be determined by doing str search? 
If this is the only way, plz put this into a separate one-liner (eg: `is_rate_limited_exeption`) since this is also used in `stack_sync_command.py`
Disclaimer: I really hate dynamic boto3 exception...

Reply:
Unfortunately only common thing I found in both `WaiterError` and `ClientError` to determine rate limit exceptions is 'Rate exceeded' string. Maybe one day there will be RateExceededError.